UI - Part 1

jduong09 opened the pull request.

ziggysauce reviewed (changes requested):
ayeee looking good so far. i just did a preliminary UI review (haven't taken a deep dive into the code yet) but here is some initial feedback:

this looks great! excellent start to getting the mobile view working better. my best suggestion is to TRY and get everything to fit without having to vertically scroll. That will be the biggest UX issue here. Horizontal scrolling is fine I think, especially for the user's hand of cards.

I would make the center pile container a little smaller so that we can move the user's content up more. We may even want to move player's 2 and 4 up more so that they're staggered with player 3 to still simulate the effect of a round table, but gaining more vertical space.

The "Your Turn" banner takes up too much space imo. I think this can be a small floating bubble somewhere else on the screen to give back some vertical real estate.

The "Finalize Turn" and "Pass Turn" can be buttons next to each other instead of stacked on top of each other. this will (1) give you more vertical real estate back and (2) avoid the user from fat figuring a pass when they meant to finalize.

ziggysauce reviewed (changes requested):
also while mobile is looking great, the layout gets all kinds of messed up on desktop:

<img width="1322" alt="Screenshot 2025-03-25 at 10 48 32 AM" src="https://github.com/user-attachments/assets/17f2f3a2-f1de-4e84-b973-3c49121787dd" />

ziggysauce commented on components/game/game.js:
```diff
@@ -1,12 +1,18 @@
 import { mapCard, icons } from '../utilities/card';
 import { aiMoves, aiPossibleCombinations, determineHardestMove } from '../utilities/ai';
 
+const slackey = Slackey({
+  weight: '400',
+  subsets: ['latin'],
+})
```
what does this do? also add a semicolon at the end

```suggestion
const slackey = Slackey({
  weight: '400',
  subsets: ['latin'],
});
```

jduong09 replied:
Was going to add a font family. Removing now.

ziggysauce commented on components/game/game.js:
```diff
@@ -1,12 +1,18 @@
 import styles from "@/app/page.module.css";
-import React, { useState, useEffect } from 'react';
-import Hand from "@/components/gameComponents/hand";
+import Hand from "@/components/hand/hand";
```
we should be consistent with double quote vs single quote usage in this file

ziggysauce commented on components/game/game.js:
```diff
@@ -135,12 +149,14 @@ const Game = () => {
           ...playerHand,
-          skipped: true
+          skipped: true,
+          roundWin: false
```
usually airbnb style eslint rules require trailing commas here. up to you though.

ziggysauce commented on components/game/game.js:
```diff
@@ -135,12 +149,14 @@ const Game = () => {
           ...playerHand,
-          skipped: false
+          skipped: false,
+          roundWin: false
```
same comment as above

ziggysauce commented on components/game/game.js:
```diff
@@ -365,13 +403,49 @@ const Game = () => {
+      roundMessage = 'P';
+    } else if (playerObj.roundWin) {
+      roundMessage = 'W';
+    } else {
```
as mentioned on the phone, this doesn't seem necessary. we can just display a momentary alert on the screen

jduong09 commented:
Resolved merge conflicts. Updated Design based on comments.

ziggysauce reviewed (changes requested):
still running into this when trying to run the app:

```
      setTimeout(() => {
        setTurnMessage(`Player ${playerTurn + 1} is thinking...`);
        setCurrentTurnCombo(move[0]);
        setComboSelect(move[0]);
        requestCombo(move[1], move[0]);
      }, 2500);
```

<img width="1326" alt="Screenshot 2025-04-11 at 6 29 53 PM" src="https://github.com/user-attachments/assets/8669ece8-6b12-494d-8ff4-79c8a28d18fb" />

jduong09 commented:
Update 4/13

- [x] Updated selected card background
- [x] Removed Combination Selection Select Tag
- [x] Added Margin between AI player and display card
- [x] Changed "Finalized Turn" & "Pass Turn" to "Play" & "Pass", respectively
- [x] Changed "Me" to "Your Hand"
- [x] Fixed jumpiness of UI during gameplay
- [x] Removed Combination Statement in Middle Pile
- [x] Added "P" badges.
- [x] Fix Bug above

jduong09 commented:
### Update 4/16/25 
- Added pass badge, made "PASSED" an alert
- Fixed bug above (bug caused by AI not playing turn when it had 3 of spades and only high card for combination)

jduong09 commented:
### Update 4/22/25
- Merged branch "issue-37--ui-header"
- Added pass badge to user

jduong09 commented:
## Update 4/24/25
- Moved user pass badge to user hand <h2>

ziggysauce reviewed (changes requested):
Changes look better! Some more minor feedback:

In mobile (and desktop), if we show an error we should add more padding between the text and the edge of the container. It should at the very least match the padding of the edge of the container with the first card

<img width="952" alt="Screenshot 2025-05-01 at 8 12 21 AM" src="https://github.com/user-attachments/assets/6cfe309b-bc16-46ee-b3c4-7b143a582f5e" />

---

The majority of people are right-handed, so I think it would actually be a slightly improved ux to put the play button on the right and the pass button on the left to favor thumb access on mobile.

<img width="801" alt="Screenshot 2025-05-01 at 8 15 11 AM" src="https://github.com/user-attachments/assets/17507c90-b30c-47ab-a2f9-e7b05b0030c8" />

---

The new pass badge looks good. For the players 2-4 I think it's fine where you have it (right next to the player) but could look better if you instead put it on the card itself like in the corner or something. Not necessary, just thinking out loud.

---

On mobile, the dropdown select for sorting is cutoff. Instead of 25%, you can make it 50% and it will fit on most mobile devices and not intrude on the new pass badge. You can make it conditionally 50% on mobile only so it doesn't look super huge on desktop (though you still want it big enough where there isn't text cutoff on desktop either)

<img width="790" alt="Screenshot 2025-05-01 at 8 17 48 AM" src="https://github.com/user-attachments/assets/e1d24350-9d09-4fe0-8ff2-4fc5f4133ba9" />
<img width="798" alt="Screenshot 2025-05-01 at 8 18 07 AM" src="https://github.com/user-attachments/assets/dbda151a-ee9e-4ec6-bea8-03aa6fb1675f" />

jduong09 commented:
### Update 5/1/25
- Added padding to invalidError div
- Swapped Play & Pass button locations
- Changed location of pass badge
- Changed width of select tag

ziggysauce reviewed (changes requested):
New pass badge looks great! Everything worked well except for these final comments then this should be good to go!

something funky happened here where when i try to play an invalid combo the error message is no longer showing for me:

<img width="1206" alt="Screenshot 2025-05-07 at 8 37 12 AM" src="https://github.com/user-attachments/assets/bd7c06e2-8317-497b-9b33-7c65c9ff88e3" />

---

During MY turn, if i have multiple cards selected, then pass, the selected cards remain through the next turn. It's not a big deal, but it might be a slightly better UX to automatically un-select all the cards for the next turn.

jduong09 commented:
## Update 5/7/25
- Resolved merge conflicts with main branch
- Added turn message when user submits invalid combo.
- Unselected previously selected cards when user passes.

ziggysauce reviewed (changes requested):
good stuff. just one last thing, on certain mobile devices the error message gets cut off:

<img width="807" alt="Screenshot 2025-05-07 at 9 14 40 PM" src="https://github.com/user-attachments/assets/8db9fadc-1e39-4d81-891b-e3813bfa9887" />

We can take away from some of the white space around the center pile of cards i think. I'd play around with various mobile device sizes to make sure the text is always visible.

After this change this PR should be good to merge.
